File: common/src/main/java/com/skm/sftp/SftpUtil.java
package com.skm.sftp;

import net.schmizz.sshj.SSHClient;
import net.schmizz.sshj.sftp.SFTPClient;
import net.schmizz.sshj.transport.verification.PromiscuousVerifier;
import java.io.IOException;


public class SftpUtil {

  public void uploadSshj(String remoteHost, String username, String password, String localPath, String remotePath) throws IOException {
    SSHClient client = new SSHClient();
    client.addHostKeyVerifier(new PromiscuousVerifier());
    client.authPassword(username, password);
    client.connect(remoteHost);
    SFTPClient sftpClient = client.newSFTPClient();
    sftpClient.put(localPath, remotePath);
    sftpClient.close();
    client.disconnect();
  }

  public void downloadSshj(String remoteHost, String username, String password, String localPath, String remotePath) throws IOException {
    SSHClient client = new SSHClient();
    client.addHostKeyVerifier(new PromiscuousVerifier());
    client.authPassword(username, password);
    client.connect(remoteHost);
    SFTPClient sftpClient = client.newSFTPClient();
    sftpClient.get(localPath, remotePath);
    sftpClient.close();
    client.disconnect();
  }
}

File: common/src/main/kotlin/com/skm/sftp/SftpUtil.kt
package com.skm.sftp


import net.schmizz.sshj.SSHClient
import net.schmizz.sshj.sftp.SFTPClient
import net.schmizz.sshj.transport.verification.PromiscuousVerifier

class SftpUtil(
    private val remoteHost: String,
    private val username: String,
    private val password: String,
    private val localPath: String?,
    private val remotePath: String?
) {

    fun uploadSshj() {
        val sshClient: SSHClient = setupSshj()
        val sftpClient: SFTPClient = sshClient.newSFTPClient()
        sftpClient.put(localPath, remotePath)
        sftpClient.close()
        sshClient.disconnect()
    }

    fun downloadSshj() {
        val sshClient: SSHClient = setupSshj()
        val sftpClient: SFTPClient = sshClient.newSFTPClient()
        sftpClient.get(remotePath, localPath)
        sftpClient.close()
        sshClient.disconnect()
    }


    private fun setupSshj(): SSHClient {
        val client = SSHClient()
        client.addHostKeyVerifier(PromiscuousVerifier())
        client.connect(remoteHost)
        client.authPassword(username, password)
        return client
    }
}